pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -126,6 +127,20 @@ func SetDefaults_Shoot(obj *Shoot) {
 		}
 		obj.Spec.Addons.KubernetesDashboard.AuthenticationMode = &defaultAuthMode
 	}
+
+	if obj.Spec.Purpose == nil {
+		p := ShootPurposeEvaluation
+
+		// backwards compatibility - take purpose from annotation if given. If not, default.
+		// TODO: This code can be removed in a future version
+		if v, ok := obj.Annotations["garden.sapcloud.io/purpose"]; ok && (v == string(ShootPurposeEvaluation) || v == string(ShootPurposeTesting) || v == string(ShootPurposeDevelopment) || v == string(ShootPurposeProduction) || (v == string(ShootPurposeInfrastructure) && obj.Namespace == v1beta1constants.GardenNamespace)) {
```

Comment:
I would firstly check for the new label key "gardener.cloud" to give it more priority compared to the old one "garden.sapcloud.io".

```suggestion
		if v, ok := obj.Annotations[v1beta1constants.GardenPurpose]; ok && (v == string(ShootPurposeEvaluation) || v == string(ShootPurposeTesting) || v == string(ShootPurposeDevelopment) || v == string(ShootPurposeProduction) || (v == string(ShootPurposeInfrastructure) && obj.Namespace == v1beta1constants.GardenNamespace)) {
```